app/helpers/restful_controller_mixin.rb

```diff
@@ -0,0 +1,16 @@
+module RestfulControllerMixin
+  def controller_restful?
+    # want to be able to cache false, so no ||=
+    return @_restful_cache unless @_restful_cache.nil?
+
+    @_restful_cache = (
+    obj = @view_binding.receiver
+
+    if obj.respond_to? :controller
+      obj.controller.try(:restful?)
+    else
+      obj.try(:restful?)
+    end
+    )
+  end
```

Comment:
Indentation is off.  But I think it can be simplified as follows:
```ruby
obj = @view_binding.receiver
@_restful_cache = if obj.respond_to? :controller
                                  obj.controller.try(:restful?)
                               else
                                  obj.try(:restful?)
                                end
```

or using ternary operator
```ruby
obj = @view_binding.receiver
@_restful_cache = obj.respond_to?(:controller) ? obj.controller.try(:restful?) : obj.try(:restful?)
```